pkg/proxyserver/controllers/clustermanagementaddon_controller.go

```diff
@@ -130,10 +130,6 @@ func (c *ClusterManagementAddonReconciler) Reconcile(ctx context.Context, reques
 	// get the related proxy configuration
 	config := &proxyv1alpha1.ManagedProxyConfiguration{}
 	if err := c.Client.Get(ctx, types.NamespacedName{Name: managedProxyConfigurationName}, config); err != nil {
-		if apierrors.IsNotFound(err) {
```

Comment:
if the managedproxyconfig is not found, we should return error, because the config might be created later or kube-apiserver may be not accessible at this moment.